x-pack/plugins/canvas/server/feature.test.ts

```diff
@@ -13,6 +13,8 @@ beforeEach(() => {
   mockReportingPlugin = {
     usesUiCapabilities: () => false,
     registerExportTypes: () => {},
+    getSpaceId: jest.fn(),
+    getScreenshots: jest.fn(),
```

Comment:
In this PR, let's create this new file called `x-pack/plugins/reporting/server/mocks/index.ts`:
```
export const reportingMock = {
  createStart: (): ReportingStart => ({
    usesUiCapabilities: () => false,
    registerExportTypes: () => {},
    getSpaceId: jest.fn(),
    getScreenshots: jest.fn(),
  }),
};
```

In this test, we'll import the test deps from Reporting:
```
import { ReportingStart } from '@kbn/reporting-plugin/server/types';
import { reportingMock } from '@kbn/reporting-plugin/server/mocks';
```

The local mock will be instantiated using the Reporting helper:
```
let mockReportingPlugin: ReportingStart;
beforeEach(() => {
  mockReportingPlugin = reportingMock.createStart();
});
```

In the "Calls on Reporting whether to include Generate PDF as a sub-feature" test, we augment the local mock with:
```
  mockReportingPlugin.usesUiCapabilities = () => true;
```